src/general/GameWorld.cs

```diff
@@ -310,6 +318,36 @@ public Species GetSpecies(uint id)
         return worldSpecies[id];
     }
 
+    /// <summary>
+    ///   Stores a description of a global event into the game world record.
+    /// </summary>
+    /// <param name="description">The event's description</param>
+    /// <param name="highlight">If true, the event will be highlighted in a timeline UI</param>
+    /// <param name="iconPath">Resource path to the icon of the event</param>
+    public void LogEvent(LocalizedString description, bool highlight = false, string iconPath = null)
+    {
+        if (eventsLog.Count > Constants.GLOBAL_EVENT_LOG_CAP)
+        {
+            var oldestKey = double.MaxValue;
+
+            foreach (var entry in eventsLog)
+                oldestKey = Math.Min(entry.Key, oldestKey);
+
+            eventsLog.Remove(oldestKey);
+        }
+
+        if (!eventsLog.ContainsKey(TotalPassedTime))
+            eventsLog.Add(TotalPassedTime, new List<GameEventDescription>());
+
+        // Event already logged in timeline
+        if (eventsLog[TotalPassedTime].Any(entry => entry.Description.ToString() == description.ToString()))
```

Comment:
I think LocalizedString might need to have a more complex comparison operator, because this won't match if the user changes language...

Reply:
A multi-level condition check would work for the comparison, mayhaps, so like 1. check if actual string match 2. check if the internal string match 3. check if reference match.

Reply:
LocalizedString should get specialized Equals method that first compares the base text and then uses Equals on all the values it has. That way it should compare equally if things are the same, not considering when the text was saved.
That would be the proper way. But actually now that I think about it converting to string *now* (as long as all substrings are LocalizedStrings as well) should result in the same thing. Though that's much less efficient as string builders and recursive string formatting will be performed.